apps/definition/test/definition_test.exs

```diff
@@ -17,28 +18,52 @@ defmodule DefinitionTest do
     end
 
     def migrate(%__MODULE__{version: 1} = old) do
-      struct(__MODULE__, %{version: 2, bar: String.to_integer(old.bar)})
+      struct(__MODULE__, %{version: 2, id: old.id, bar: String.to_integer(old.bar)})
```

Comment:
this will cause an `KeyError` exception if you pass in a version of the struct that was created before an id was a required field and since it hasn't been "newed" into a struct yet there's not default value (like nil) to fall back on.

Reply:
I think I've addressed your concern about this, but let me know if this doesn't cut it.